### ISSUE TITLE
NullPointerException in ConfigBeanImpl.isOptionalProperty when using ConfigBeanFactory.create

### ISSUE THREAD

**Issue by ScalaWilliam** [typesafehub/config#447]
Thanks to Viktor Klang, discovered ConfigBeanFactory but it's not quite working against an S3 Client Configuration.

build.sbt:
```scala
libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.11.73"
libraryDependencies += "com.typesafe" % "config" % "1.3.1"
```

FailApp.scala:
```scala
import com.amazonaws.ClientConfiguration
import com.typesafe.config.{ConfigBeanFactory, ConfigFactory}

object FailApp extends App {
  ConfigBeanFactory.create(ConfigFactory.empty(), classOf[ClientConfiguration])
}
```

Running:
```bash 
sbt run
```

Result:
```
java.lang.NullPointerException
	at com.typesafe.config.impl.ConfigBeanImpl.isOptionalProperty(ConfigBeanImpl.java:280)
	at com.typesafe.config.impl.ConfigBeanImpl.createInternal(ConfigBeanImpl.java:95)
	at com.typesafe.config.ConfigBeanFactory.create(ConfigBeanFactory.java:47)
	at FailApp$delayedInit$body.apply(FailApp.scala:5)
```

Expectation:
```
No exception, no output.
```

Here's a gist, you can just clone it using the command in the comment:
https://gist.github.com/ScalaWilliam/6f267928f222e4ac80f43bb0d663d871#gistcomment-1958918

**Comment by havocp** [comment]
The issue appears to be that there's no field called `userAgent`, but there are methods `getUserAgent`/`setUserAgent`. It seems the user agent is deprecated in favor of getUserAgentPrefix/Suffix, so maybe the user agent field is now "virtual" (defined in terms of those other properties).

The easy fix could be to modify this block:
```
            List<PropertyDescriptor> beanProps = new ArrayList<PropertyDescriptor>();
            for (PropertyDescriptor beanProp : beanInfo.getPropertyDescriptors()) {
                if (beanProp.getReadMethod() == null || beanProp.getWriteMethod() == null) {
                    continue;
                }
                beanProps.add(beanProp);
            }
```

Which ignores props without a get or set, to also ignore props without a field?

A harder fix might be, if there's no field, invoke the getter/setter instead, but maybe that's overkill.

PRs welcome, ideally with regression test case. :-) Otherwise timeline is unknown but should happen eventually.

Thanks for the report!

**Comment by 2m** [comment]
Fixed in #454
